apis/telemetry/v1alpha1/telemetrymanager_types.go

```diff
@@ -0,0 +1,131 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+type State string
+
+var (
+	ConditionTypeInstallation = "Installation"
+	ConditionReasonReady      = "Ready"
+)
+
+// Valid Module CR States.
+const (
+	// StateReady signifies Module CR is Ready and has been installed successfully.
+	StateReady State = "Ready"
+
+	// StateProcessing signifies Module CR is reconciling and is in the process of installation.
+	// Processing can also signal that the Installation previously encountered an error and is now recovering.
+	StateProcessing State = "Processing"
+
+	// StateError signifies an error for Module CR. This signifies that the Installation
+	// process encountered an error.
+	// Contrary to Processing, it can be expected that this state should change on the next retry.
+	StateError State = "Error"
+
+	// StateDeleting signifies Module CR is being deleted. This is the state that is used
+	// when a deletionTimestamp was detected and Finalizers are picked up.
+	StateDeleting State = "Deleting"
+)
+
+// TelemetryManagerSpec defines the desired state of TelemetryManager
+type TelemetryManagerSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
+
+// TelemetryManagerStatus defines the observed state of TelemetryManager
+type TelemetryManagerStatus struct {
+	Status `json:",inline"`
+
+	// Conditions contain a set of conditionals to determine the State of Status.
+	// If all Conditions are met, State is expected to be in StateReady.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// add other fields to status subresource here
+}
+
+func (s *TelemetryManagerStatus) WithState(state State) *TelemetryManagerStatus {
+	s.State = state
+	return s
+}
+
+func (s *TelemetryManagerStatus) WithInstallConditionStatus(status metav1.ConditionStatus, objGeneration int64) *TelemetryManagerStatus {
+	if s.Conditions == nil {
+		s.Conditions = make([]metav1.Condition, 0, 1)
+	}
+
+	condition := meta.FindStatusCondition(s.Conditions, ConditionTypeInstallation)
+
+	if condition == nil {
+		condition = &metav1.Condition{
+			Type:    ConditionTypeInstallation,
+			Reason:  ConditionReasonReady,
+			Message: "installation is ready and resources can be used",
+		}
+	}
+
+	condition.Status = status
+	condition.ObservedGeneration = objGeneration
+	meta.SetStatusCondition(&s.Conditions, *condition)
+	return s
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+// TelemetryManager is the Schema for the telemetrymanagers API
+type TelemetryManager struct {
```

Comment:
don't forget to rename it to `Telemetry` as well as the other types

Reply:
renamed